### PR TITLE
Refactor StopTheLineJenkins git hook

### PR DESCRIPTION
**Pull request by nadeemahmad**
The original `StopTheLineJenkins` `HookAction` was using deprecated code and configuration framework. This change updates the git hook to use the new configuration framework and the recently updated `Jenkins\Job` class.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @nadeemahmad, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: src/Bart/GitHook/GitHookConfig.php [hunk 1]
```diff
@@ -29,6 +29,12 @@ public function README()
 ; %s will be replaced with commit revision hash
 comment_template = 'Commit %s pushed to JIRA. See online at https://git.example.com/?h=%s'
 
+[jenkins]
+; Used by StopTheLineJenkins Hook Action to allow
+; commits that have this directive to go through,
+; when the build is broken. Defaults to {buildfix}.
+build_fix_directive = '{buildfix}'
```
Comment (asheepapart) [inline src/Bart/GitHook/GitHookConfig.php]: This is cool, but I think more than necessary. What do you think about,

```php
# src/Bart/GitHook/Directives.php

namespace Bart\GitHook;

final class Directives extends Enumerations ... {
     val buildfix = "buildfix";
}

File: src/Bart/Jenkins/JenkinsConfig.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        return $this->getValue('api', 'protocol', 'http', false);
+    }
+
+    /**
+     * @return string The port that the Jenkins instance is running on
+     * @throws ConfigurationException
+     */
+    public function port()
+    {
+        return $this->getValue('api', 'port', 8080, false);
```
Comment (asheepapart) [inline src/Bart/Jenkins/JenkinsConfig.php]: You say `@return string`, but the default is an integer.

Comment (nadeemahmad) [inline src/Bart/Jenkins/JenkinsConfig.php]: Done.

File: src/Bart/GitHook/Directives.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * Class Directives
+ * @package Bart\GitHook
+ */
+class Directives extends AbstractEnumeration
+{
+    /**
+     * Used by StopTheLineJenkins Hook Action to allow commits that have this
+     * directive to go through, when the build is broken.
+     */
+    const buildFix = '{buildFix}';
```
Comment (asheepapart) [inline src/Bart/GitHook/Directives.php]: This should be upper case,  `BUILD_FIX`

Comment (nadeemahmad) [inline src/Bart/GitHook/Directives.php]: Right, of course.

**Pull request merged**